spec/ysh-proc.test.sh

```diff
@@ -524,3 +524,28 @@ grep
 ## STDOUT:
 sh-func grep
 ## END
+
+#### proc resolution changes with the local scope
```

Comment:
I'm not 100% sure why we need this

It seems reasonable and consistent, but maybe add some comments about what goes wrong if we don't have it?